pkg/runner/runner.go

```diff
@@ -602,7 +604,55 @@ func (r Runner) Update() error {
 	return nil
 }
 
-//Generate new deployer
+// Refresh will refresh autoscaling group instances
+func (r Runner) Refresh() error {
+	i := inspector.New(r.Builder.Config.Region)
+
+	asg, err := i.SelectStack(r.Builder.Config.Application)
+	if err != nil {
+		return err
+	}
+	r.Logger.Debugf("Selected target autoscaling group: %s", asg)
+
+	group, err := i.GetStackInformation(asg)
+	if err != nil {
+		return err
+	}
+
+	r.Logger.Debugf("Autoscaling group found: %s", *group.AutoScalingGroupARN)
+
+	if err := tool.LocalCheck("Do you really want to refresh? ", r.Builder.Config.AutoApply); err != nil {
+		return err
+	}
+
+	r.Logger.Debug("Create a new refresher")
+	refresher := refresh.New(r.Builder.Config.Region)
+	refresher.SetTarget(group)
+
+	input := make(chan error)
+	go func(ch chan error) {
+		ch <- refresher.StartRefresh(r.Builder.Config.InstanceWarmup, r.Builder.Config.MinHealthyPercentage)
+	}(input)
+
+	time.Sleep(3 * time.Second)
+
+	if err := refresher.StatusCheck(r.Builder.Config.PollingInterval, r.Builder.Config.Timeout); err != nil {
+		return err
+	}
+
+	if err := refresher.PrintResult(); err != nil {
+		return err
+	}
+
+	//if err := <-input; err != nil {
```

Comment:
I think this code can check StartRefresh function's result when it returned error.
Isn't it?

Reply:
Yes, I think I rather uncomment this.

I commented this because it could be confusing because triggering instance refresh does not finish until all processes are done. This means that I cannot do the health check after triggering the job. 
So I implemented triggering part with go routine so that I can check if the process is finished or not after triggering. 
It means that even though the job is finished without error, but if there is an error in triggering part, users get an error message. 

I think I need to look into this more.

Reply:
Thank you for your explanation. I've understood.